evi-react-native-example/App.tsx

```diff
@@ -0,0 +1,392 @@
+import React, { useEffect, useState, useRef } from "react";
+import {
+  View,
+  Text,
+  Button,
+  StyleSheet,
+  ScrollView,
+  SafeAreaView,
+  LayoutAnimation,
+} from "react-native";
+import {useEvent} from 'expo'
+
+// We use Hume's low-level typescript SDK for this example.
+// The React SDK (@humeai/voice-react) does not support React Native.
+import { HumeClient, type Hume } from "hume";
+
+// An expo native module is included with this example to handle audio
+// recording and playback. While some react-native libraries are available,
+// none both provide a streaming interface and support for enabling echo
+// cancellation, which is necessary for a good user experience with EVI.
+//
+// The provided native module is a good starting place, but you should
+// modify it to fit the audio recording needs of your specific app.
+import NativeAudio, { AudioEventPayload } from "./modules/audio";
+
+// Represents a chat message in the chat display.
+interface ChatEntry {
+  role: "user" | "assistant";
+  timestamp: string;
+  content: string;
+}
+
+
+/**
+ * An AudioClip is a function that you can call to play some audio.
+ * It returns a promise that is resolved when the audio is finished playing.
+ */
+type AudioClip = () => Promise<void>;
+
+// EVI can send audio output messages faster than they can be played back.
+// It is important to buffer them in a queue so as not to cut off a clip of
+// playing audio with a more recent clip. `audioQueue` is a global
+// audio queue that manages this buffering.
+const audioQueue = {
```

Comment:
Nit/optional: We did it as a performance consideration in the consumer app, but might consider moving the audio queue logic into native and have the websocket handler just publish messages to the native queue. We found that running in bridge mode ("old architecture") promises could be quite slow to resolve from native -> JS. It's possible this isn't an issue with the Expo/swift combo but thought I'd point it out just in case

Reply:
Done in [Enqueue audio on the swift side!](https://github.com/HumeAI/hume-api-examples/pull/100/commits/e8860647386f815409be0ff3f44c6c0feeadda93)